src/components/inputs/DynamicField/DynamicField.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useCallback, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import { Autocomplete, DateTime, TextField } from 'components'
+import { ControlType, DynamicFieldProps } from './types'
+import { Checkbox } from '@mui/material'
+import { FormControlLabel } from '@mui/material'
+
+function DynamicField<TCustomComponentProps extends object = any, TAutocompleteOptions = any>(
+  props: DynamicFieldProps<TCustomComponentProps, TAutocompleteOptions>
+) {
+  const {
+    controlType,
+    id,
+    value,
+    label,
+    onChange,
+    options,
+    loadOptions,
+    error,
+    helperText,
+    CustomComponent,
+    customComponentProps,
+    ...rest
+  } = props
+
+  const onCheckboxChange = useCallback(
+    (_: React.ChangeEvent<HTMLInputElement>, checked: boolean) => onChange(checked),
+    [onChange]
+  )
+
+  const defaultControl = useMemo(
+    () => <TextField id={id} value={String(value)} label={label} error={error} helperText={helperText} disabled />,
+    [id, value, label, error, helperText]
+  )
+
+  switch (controlType) {
+    case ControlType.Text:
+      return (
+        <TextField
+          id={id}
+          value={(value as string | number) || ''}
+          label={label}
+          onChange={onChange}
+          fullWidth
+          error={error}
+          helperText={helperText}
+          {...rest}
+        />
+      )
+    case ControlType.Integer:
+      return (
+        <TextField
+          id={id}
+          value={(value as string | number) || ''}
```

Comment:
you can use emptyString constant defined in src/testingUtils

Reply:
Mi se pare ciudata chestia asta pe care o avem cu emptyString, emptyArray, etc. Mi se pare mai lizibil sa pui constanta. Dar o sa pun constanta. 
Ca nota aici: nu cred ca ar trebui sa fie nevoie sa punem || emptyString. Si chiar apar probleme daca nu punem. Cred ca componentele astea ar trebui sa gestioneze singure cazurile astea.